src/utils/async-queue.ts

```diff
@@ -0,0 +1,42 @@
+export type QueueEntry = () => void;
+
+// queue manager is an class that helps with managing tasks.
+// More specifically, it helps with tasks that are often used.
+// It's fully asyncronous and uses promises and tries to get 60FPS.
+export default class AsyncQueue {
+    private queue: QueueEntry[] = [];
+    private timerID: number = null;
+    private frameDuration = 1000 / 60;
+
+    addToQueue(entry: QueueEntry) {
+        this.queue.push(entry);
+        this.requestQueue();
+    }
+
+    stopQueue() {
+        if (this.timerID !== null) {
+            cancelAnimationFrame(this.timerID);
+            this.timerID = null;
+        }
+        this.queue = [];
+    }
+
+    // Ensures 60FPS.
+    private requestQueue() {
```

Comment:
Something is wrong with the name `requestQueue`. It requests a queue in a queue :)

But actually it runs the tasks in synchronous chunks, until they start causing a FPS drop. So, try some another short and descriptive name.

Not sure if it is a problem in this case, but the first chunk is not run synchronously. You can leave it as is, but if a flash of images will be noticeable, maybe it could be improved by running the first chunk synchronously later.

Reply:
Hopefully that won't happend the flash of images. As another workaround would be to temporialy have `background-image: none` on those selectors and have asyncRebuild add the correctly modified value.